curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -117,6 +120,7 @@ public void processResult(CuratorFramework client, CuratorEvent event)
             {
                 //Update is ok, mark initialisation as complete if required.
                 initialisationComplete();
+                notifyListeners();
```

Comment:
@lvfangmin I looked at this - and it looks to me like this is the callback where the result of node creation comes back (telling us whether it was successful, node already exists, etc). This is in the `processBackgroundCallback` method, and where we know that the result code comes back as `false` for `KeeperException.Code.NODEEXISTS` and `true` for  `KeeperException.Code.OK`.  This means that we just created the node, and seemingly the place to notify listeners.

The callback for updating the node value seems to be `setDataCallback`, which I did not put any code for notifying listeners.

To me this looked like the place to put this, but definitely let me know if you thought of a better place and I can investigate that.